### PR TITLE
python3Packages.typed-ast: 1.4.1 -> 1.4.3

### PR DESCRIPTION
**Pull request by lukegb**
###### Motivation for this change

https://github.com/python/typed_ast

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/typed-ast/default.nix [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
     rev = version;
-    sha256 = "086r9qhls6mz1w72a6d1ld3m4fbkxklf6mgwbs8wpw0zlxjm7y40";
+    sha256 = "16mn9snwik5n2ib65sw2xcaqdm02j8ps21zgjxf8kyy7qnx2mx4w";
   };
   # Only works with Python 3.3 and newer;
   disabled = pythonOlder "3.3";
   # No tests in archive
   doCheck = false;
   meta = {
     homepage = "https://pypi.python.org/pypi/typed-ast";
```
Comment (dotlambda) [inline pkgs/development/python-modules/typed-ast/default.nix:16]:
```suggestion
    homepage = "https://github.com/python/typed_ast";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please add `pythonImportsCheck`.

File: pkgs/development/python-modules/typed-ast/default.nix (hunk 1 above)
Comment (lukegb) [reply pkgs/development/python-modules/typed-ast/default.nix]: Done, thanks.

**Comment by lukegb** [comment]
> Please add `pythonImportsCheck`.

Done.

I also added pytest, but pytestCheckHook doesn't work for this for... somewhat convoluted sys.path reasons.

**Comment by lukegb** [comment]
@ofborg build python3.pkgs.typed-ast

**Comment by dotlambda** [comment]
> but pytestCheckHook doesn't work for this for

Doing `rm -r typed_ast` in `preCheck` should do the trick.

**Comment by lukegb** [comment]
> > but pytestCheckHook doesn't work for this for
> 
> Doing `rm -r typed_ast` in `preCheck` should do the trick.

I considered that too -- the other thing I considered was changing directory up one level in preCheck, but... in terms of fragility I prefer this approach, I think?

File: pkgs/development/python-modules/typed-ast/default.nix [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+
+    runHook postCheck
+  '';
+
   meta = {
-    homepage = "https://pypi.python.org/pypi/typed-ast";
-    description = "a fork of Python 2 and 3 ast modules with type comment support";
+    homepage = "https://github.com/python/typed_ast";
+    description = "Python 2 and 3 ast modules with type comment support";
     license = lib.licenses.asl20;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/typed-ast/default.nix:38]: Maybe you could add yourself as a maintainer.

**Pull request merged**